python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -219,6 +226,11 @@ def __init__(
                 )
             self._assets_by_key[asset_key] = assets_def
 
+        self._partitions_def_by_asset_key = {
+            asset_key: asset_def.partitions_def
```

Comment:
if an asset isn't partitioned will `asset_def.partitions_def` be None?

Reply:
Yep, we raise an error in every time we try to fetch a partitions definition but it is None